### PR TITLE
[INLONG-7089][Manager] Separate the concept of node tag from the node table and extract the concept of task group

### LINKED ISSUE #7089: [Improve][Manager] Separate the concept of node tag from the node table and extract the concept of task group
### Description

Separate the concept of node tag from the node table and extract the concept of task group

### InLong Component

InLong Manager

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by thesumery**
[INLONG-7089][Agent] Separate the concept of node tag from the node table and extract the concept of task group

### Prepare a Pull Request
*(Change the title refer to the following example)*

- Title Example: [INLONG-7089][Manager] Separate the concept of node tag from the node table and extract the concept of task group
- Fixes #7089 

### Motivation

*Separate the concept of node tag from the node table and extract the concept of task group*

### CODE REVIEW

File: inlong-manager/manager-dao/src/main/java/org/apache/inlong/manager/dao/mapper/InlongLabelEntityMapper.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.inlong.manager.dao.mapper;
+
+import org.apache.inlong.manager.dao.entity.InlongLabelEntity;
+
```
Comment (fuweng11) [inline inlong-manager/manager-dao/src/main/java/org/apache/inlong/manager/dao/mapper/InlongLabelEntityMapper.java:21]: Add`@Repository`.

File: inlong-manager/manager-dao/src/main/java/org/apache/inlong/manager/dao/entity/InlongLabelEntity.java [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.inlong.manager.dao.entity;
+
+import java.io.Serializable;
+import java.util.Date;
+
+public class InlongLabelEntity implements Serializable {
```
Comment (fuweng11) [inline inlong-manager/manager-dao/src/main/java/org/apache/inlong/manager/dao/entity/InlongLabelEntity.java:23]: Suggest to use `@Data`, `@EqualsAndHashCode(callSuper = true)`, etc

File: inlong-manager/manager-dao/src/main/java/org/apache/inlong/manager/dao/entity/InlongLabelNodeRelationEntity.java [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.inlong.manager.dao.entity;
+
+import java.io.Serializable;
+
+public class InlongLabelNodeRelationEntity implements Serializable {
```
Comment (fuweng11) [inline inlong-manager/manager-dao/src/main/java/org/apache/inlong/manager/dao/entity/InlongLabelNodeRelationEntity.java:22]: Suggest to use `@Data`, `@EqualsAndHashCode(callSuper = true)`, etc

File: inlong-manager/manager-dao/src/main/java/org/apache/inlong/manager/dao/mapper/InlongLabelNodeRelationEntityMapper.java [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+ * limitations under the License.
+ */
+
+package org.apache.inlong.manager.dao.mapper;
+
+import org.apache.ibatis.annotations.Param;
+import org.apache.inlong.manager.dao.entity.InlongLabelNodeRelationEntity;
+
+import java.util.List;
+
```
Comment (fuweng11) [inline inlong-manager/manager-dao/src/main/java/org/apache/inlong/manager/dao/mapper/InlongLabelNodeRelationEntityMapper.java:24]: Add`@Repository.`

File: inlong-manager/manager-dao/src/main/resources/mappers/InlongLabelEntityMapper.xml [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+    <result column="version" jdbcType="INTEGER" property="version" />
+  </resultMap>
+  <resultMap extends="BaseResultMap" id="ResultMapWithBLOBs" type="org.apache.inlong.manager.dao.entity.InlongLabelEntity">
+    <result column="ext_params" jdbcType="LONGVARCHAR" property="extParams" />
+  </resultMap>
+  <sql id="Base_Column_List">
+    id, label_name, description, in_charges, is_deleted, creator, modifier, create_time, 
+    modify_time, version
+  </sql>
+  <sql id="Blob_Column_List">
```
Comment (fuweng11) [inline inlong-manager/manager-dao/src/main/resources/mappers/InlongLabelEntityMapper.xml:42]: Why `ext_params` are not put together with other parameters？

File: inlong-manager/manager-dao/pom.xml [hunk 6]
```diff
@@ -114,6 +114,11 @@
                         <artifactId>postgresql</artifactId>
                         <version>${postgresql.version}</version>
                     </dependency>
+                    <dependency>
```
Comment (gosonzhang) [inline inlong-manager/manager-dao/pom.xml:117]: mysql-connector-java is LGPL, NEED PROVIDED

File: inlong-manager/manager-web/src/main/java/org/apache/inlong/manager/web/controller/InlongClusterController.java [hunk 7]
```diff
@@ -228,11 +228,11 @@ public Response<Boolean> deleteNode(@PathVariable Integer id) {
         return Response.success(clusterService.deleteNode(id, LoginUserUtils.getLoginUser().getName()));
     }
 
-    @RequestMapping(value = "/cluster/node/bindTag")
+    @RequestMapping(value = "/cluster/node/bindLabel")
```
Comment (gosonzhang) [inline inlong-manager/manager-web/src/main/java/org/apache/inlong/manager/web/controller/InlongClusterController.java:231]: Should the operation of binding the label be a separate operation class?

It is only associated with the file agent

File: inlong-manager/manager-web/sql/apache_inlong_manager.sql [hunk 8]
```diff
@@ -158,6 +158,40 @@ CREATE TABLE IF NOT EXISTS `inlong_cluster_node`
 ) ENGINE = InnoDB
   DEFAULT CHARSET = utf8mb4 COMMENT ='Inlong cluster node table';
 
+-- ----------------------------
+-- Table structure for inlong_label
+-- ----------------------------
+CREATE TABLE IF NOT EXISTS `inlong_label`
```
Comment (gosonzhang) [inline inlong-manager/manager-web/sql/apache_inlong_manager.sql:164]: Label is a word with no clear meaning. Can you give it a clear name according to the characteristics?

**Comment by dockerzhang** [comment]
Please update https://github.com/apache/inlong/blob/master/inlong-manager/manager-web/sql/changes-1.5.0.sql at the same time.

File: inlong-manager/manager-web/src/main/java/org/apache/inlong/manager/web/controller/InlongClusterController.java (hunk 7 above)
Comment (thesumery) [reply inlong-manager/manager-web/src/main/java/org/apache/inlong/manager/web/controller/InlongClusterController.java:231]: At present, the label is not bound to the file agent forcibly, and its association is also bound to the node. 

And the design of the label is not just for collection tasks

File: inlong-agent/agent-core/src/main/java/org/apache/inlong/agent/core/HeartbeatManager.java [hunk 9]
```diff
@@ -228,7 +228,7 @@ private HeartbeatMsg buildHeartbeatMsg() {
             heartbeatMsg.setInCharges(inCharges);
         }
         if (StringUtils.isNotBlank(nodeTag)) {
-            heartbeatMsg.setNodeTag(nodeTag);
+            heartbeatMsg.setNodeLabel(nodeTag);
```
Comment (gosonzhang) [inline inlong-agent/agent-core/src/main/java/org/apache/inlong/agent/core/HeartbeatManager.java:231]: Can there be a clear meaning of the name?

tag and lab are too general

File: inlong-manager/manager-web/src/main/java/org/apache/inlong/manager/web/controller/InlongClusterController.java (hunk 7 above)
Comment (gosonzhang) [reply inlong-manager/manager-web/src/main/java/org/apache/inlong/manager/web/controller/InlongClusterController.java:231]: bindNodeLabel, not bindNodeTag

File: inlong-agent/agent-core/src/main/java/org/apache/inlong/agent/core/HeartbeatManager.java (hunk 9 above)
Comment (thesumery) [reply inlong-agent/agent-core/src/main/java/org/apache/inlong/agent/core/HeartbeatManager.java:231]: Table name has  alreay  embodied its meaning

File: inlong-manager/manager-pojo/src/main/java/org/apache/inlong/manager/pojo/cluster/ClusterNodeBindLabelRequest.java [hunk 10]
```diff
@@ -29,11 +29,11 @@
  */
 @Data
 @ApiModel("Cluster node bind and unbind tag request")
-public class ClusterNodeBindTagRequest {
+public class ClusterNodeBindLabelRequest {
 
     @NotBlank(message = "Cluster nodeTag cannot be blank")
     @ApiModelProperty(value = "Cluster node tag")
```
Comment (fuweng11) [inline inlong-manager/manager-pojo/src/main/java/org/apache/inlong/manager/pojo/cluster/ClusterNodeBindLabelRequest.java:35]: Add explanation above. like `@ApiModelProperty(value = "MQ resource", notes = "in inlong group, TubeMQ corresponds to Topic, Pulsar corresponds to Namespace")`

File: inlong-manager/manager-pojo/src/main/java/org/apache/inlong/manager/pojo/source/SourceRequest.java [hunk 11]
```diff
@@ -72,8 +72,8 @@ public class SourceRequest {
     @ApiModelProperty("Inlong cluster name")
     private String inlongClusterName;
 
-    @ApiModelProperty("Inlong cluster node tag")
-    private String inlongClusterNodeTag;
+    @ApiModelProperty("Inlong cluster node label")
```
Comment (fuweng11) [inline inlong-manager/manager-pojo/src/main/java/org/apache/inlong/manager/pojo/source/SourceRequest.java:75]: Add explanation above. like @ApiModelProperty(value = "MQ resource", notes = "in inlong group, TubeMQ corresponds to Topic, Pulsar corresponds to Namespace").

File: inlong-common/src/main/java/org/apache/inlong/common/heartbeat/HeartbeatMsg.java [hunk 12]
```diff
@@ -78,7 +78,7 @@ public class HeartbeatMsg {
     /**
       * Tag of node, separated by commas(,)
```
Comment (fuweng11) [inline inlong-common/src/main/java/org/apache/inlong/common/heartbeat/HeartbeatMsg.java:79]: Uniform annotation and method name.

File: inlong-manager/manager-client/src/main/java/org/apache/inlong/manager/client/api/inner/client/InlongClusterClient.java [hunk 13]
```diff
@@ -304,7 +304,7 @@ public Boolean deleteNode(Integer id) {
      * @param request cluster info to be modified
```
Comment (fuweng11) [inline inlong-manager/manager-client/src/main/java/org/apache/inlong/manager/client/api/inner/client/InlongClusterClient.java:304]: Uniform annotation and method name.

File: inlong-manager/manager-client/src/main/java/org/apache/inlong/manager/client/api/service/InlongClusterApi.java [hunk 14]
```diff
@@ -101,5 +101,5 @@ Call<Response<List<ClusterNodeResponse>>> listNodeByGroupId(@Query("inlongGroupI
     Call<Response<Boolean>> deleteNode(@Path("id") Integer id);
 
     @POST("cluster/node/bindTag")
```
Comment (fuweng11) [inline inlong-manager/manager-client/src/main/java/org/apache/inlong/manager/client/api/service/InlongClusterApi.java:103]: Uniform url and method name.

File: inlong-manager/manager-dao/src/main/java/org/apache/inlong/manager/dao/entity/StreamSourceLabelEntity.java [hunk 15]
```diff
@@ -0,0 +1,54 @@
...
+
+package org.apache.inlong.manager.dao.entity;
+
+import lombok.Data;
+
+import java.io.Serializable;
+import java.util.Date;
+
+/**
+ * Inlong label entity.
```
Comment (fuweng11) [inline inlong-manager/manager-dao/src/main/java/org/apache/inlong/manager/dao/entity/StreamSourceLabelEntity.java:26]: Unified annotation and class name.

File: inlong-manager/manager-dao/src/main/java/org/apache/inlong/manager/dao/entity/StreamSourceLabelNodeRelationEntity.java [hunk 16]
```diff
@@ -0,0 +1,37 @@
...
+ */
+
+package org.apache.inlong.manager.dao.entity;
+
+import lombok.Data;
+
+import java.io.Serializable;
+
+/**
+ * Inlong label node relation entity.
```
Comment (fuweng11) [inline inlong-manager/manager-dao/src/main/java/org/apache/inlong/manager/dao/entity/StreamSourceLabelNodeRelationEntity.java:25]: Unified annotation and class name.

File: inlong-manager/manager-pojo/src/main/java/org/apache/inlong/manager/pojo/cluster/ClusterNodeBindLabelRequest.java [hunk 17]
```diff
@@ -29,11 +29,11 @@
  */
 @Data
 @ApiModel("Cluster node bind and unbind tag request")
```
Comment (fuweng11) [inline inlong-manager/manager-pojo/src/main/java/org/apache/inlong/manager/pojo/cluster/ClusterNodeBindLabelRequest.java:31]: Unified annotation and method name

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/cluster/InlongClusterService.java [hunk 18]
```diff
@@ -393,7 +393,7 @@ List<ClusterNodeResponse> listNodeByGroupId(
      * @param operator current operator
```
Comment (fuweng11) [inline inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/cluster/InlongClusterService.java:393]: Uniform annotation and method name.

File: inlong-manager/manager-client/src/main/java/org/apache/inlong/manager/client/api/service/InlongClusterApi.java (hunk 14 above)
Comment (thesumery) [reply inlong-manager/manager-client/src/main/java/org/apache/inlong/manager/client/api/service/InlongClusterApi.java:103]: Good issue

File: inlong-manager/manager-test/src/main/resources/h2/apache_inlong_manager.sql [hunk 19]
```diff
@@ -330,7 +330,7 @@ CREATE TABLE IF NOT EXISTS `stream_source`
     `uuid`                varchar(30)           DEFAULT NULL COMMENT 'Mac uuid of the agent running the task',
     `data_node_name`      varchar(128)          DEFAULT NULL COMMENT 'Node name, which links to data_node table',
     `inlong_cluster_name` varchar(128)          DEFAULT NULL COMMENT 'Cluster name of the agent running the task',
-    `inlong_cluster_node_tag` varchar(512)      DEFAULT NULL COMMENT 'Cluster node tag',
+    `inlong_cluster_node_group` varchar(512)      DEFAULT NULL COMMENT 'Cluster node label',
```
Comment (gosonzhang) [inline inlong-manager/manager-test/src/main/resources/h2/apache_inlong_manager.sql:333]: label -- > group

File: inlong-manager/manager-web/sql/apache_inlong_manager.sql [hunk 20]
```diff
@@ -347,7 +347,7 @@ CREATE TABLE IF NOT EXISTS `stream_source`
     `uuid`                varchar(30)           DEFAULT NULL COMMENT 'Mac uuid of the agent running the task',
     `data_node_name`      varchar(128)          DEFAULT NULL COMMENT 'Node name, which links to data_node table',
     `inlong_cluster_name` varchar(128)          DEFAULT NULL COMMENT 'Cluster name of the agent running the task',
-    `inlong_cluster_node_tag` varchar(512)      DEFAULT NULL COMMENT 'Cluster node tag',
+    `inlong_cluster_node_group` varchar(512)      DEFAULT NULL COMMENT 'Cluster node label',
```
Comment (gosonzhang) [inline inlong-manager/manager-web/sql/apache_inlong_manager.sql:350]: label --> group

File: inlong-manager/manager-service/src/test/java/org/apache/inlong/manager/service/core/impl/AgentServiceTest.java [hunk 21]
```diff
@@ -310,15 +310,15 @@ public void testRematchedWhenSuspend() {
...
+        bindGroup(true, "tag1");
 
         agent.pullTask();
         agent.pullTask(); // report last success status
 
         // suspend and restart
         suspendSource(groupStream.getLeft(), groupStream.getRight());
         restartSource(groupStream.getLeft(), groupStream.getRight());
-        bindTag(false, "tag1");
+        bindGroup(false, "tag1");
```
Comment (gosonzhang) [inline inlong-manager/manager-service/src/test/java/org/apache/inlong/manager/service/core/impl/AgentServiceTest.java:321]: tag1 --> group1

File: inlong-manager/manager-service/src/test/java/org/apache/inlong/manager/service/core/impl/AgentServiceTest.java [hunk 22]
```diff
@@ -203,8 +203,8 @@ public void testTagMatch() {
         saveSource("tag2,tag3");
         saveSource("tag2,tag3");
         saveSource("tag4");
-        bindTag(true, "tag1");
-        bindTag(true, "tag2");
+        bindGroup(true, "tag1");
```
Comment (gosonzhang) [inline inlong-manager/manager-service/src/test/java/org/apache/inlong/manager/service/core/impl/AgentServiceTest.java:206]: tag1 -- > group1

**Review by fuweng11 — APPROVED** [review]

**Review by gosonzhang — APPROVED** [review]

**Review by dockerzhang — APPROVED** [review]

**Pull request merged**